### PR TITLE
iio:iio.c: Add cyclic mode size checks in pop scan

### PR DESCRIPTION
**Pull request by ribdp**
## Pull Request Description

1. Use no_os_cb_read even in cyclic mode to check if there are samples available to be read
2. Refactor the function implementation

## PR Type
- [x] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: iio/iio.c [hunk 1]
```diff
@@ -1423,18 +1423,16 @@ int iio_buffer_pop_scan(struct iio_buffer *buffer, void *data)
 	if (!buffer)
 		return -EINVAL;
 
-	if(!buffer->cyclic_info.is_cyclic)
-		return no_os_cb_read(buffer->buf, data, buffer->bytes_per_scan);
+	uint32_t ret;
```
Comment (rbolboac) [inline iio/iio.c:1426]: should be int,, otherwise lgtm

Comment (ribdp) [reply iio/iio.c]: Fixed this now

**Comment by buha** [comment]
I think @rbolboac meant 'int', not 'int32' to match the function return type.

**Comment by rbolboac** [comment]
> I think @rbolboac meant 'int', not 'int32' to match the function return type.

Yes, so I think a while ago we decided to use int instead of int32_t for new APIs so it will be best to use int also in this case, even though the result is the same because int = int32_t

**Comment by ribdp** [comment]
Changelog - 
1. Changed int32_t -> int

**Review by rbolboac — APPROVED** [review]

**Comment by ribdp** [comment]
is the PR good to be merged?

**Pull request merged**